src/Microsoft.DotNet.Wpf/src/PresentationFramework/Microsoft/Win32/OpenFolderDialog.cs

```diff
@@ -0,0 +1,215 @@
+
+        #endregion Constructors
+
+        //---------------------------------------------------
+        //
+        // Public Methods
+        //
+        //---------------------------------------------------
+        #region Public Methods
+
+        //   We override the FileDialog implementation to set a default
+        //   for OFN_FILEMUSTEXIST in addition to the other option flags
+        //   defined in FileDialog.
+        /// <summary>
+        ///  Resets all properties to their default values.
+        /// </summary>
+        /// <Remarks>
+        ///     Callers must have FileIOPermission(PermissionState.Unrestricted) to call this API.
+        /// </Remarks>
+        public override void Reset()
+        {
+
+            // it is VERY important that the base.reset() call remain here
+            // and be located at the top of this function.
+            // Since most of the initialization for this class is actually
+            // done in the FileDialog class, we must call that implementation
+            // before we can finish with the Initialize() call specific to our
+            // derived class.
+            base.Reset();
+
+            Initialize();
+        }
+
+        #endregion Public Methods
+
+        //---------------------------------------------------
+        //
+        // Public Properties
+        //
+        //---------------------------------------------------
+        //#region Public Properties
+        //#endregion Public Properties
+
+        //---------------------------------------------------
+        //
+        // Public Events
+        //
+        //---------------------------------------------------
+        //#region Public Events
+        //#endregion Public Events
+
+        //---------------------------------------------------
+        //
+        // Protected Methods
+        //
+        //---------------------------------------------------
+        #region Protected Methods
+
+        /// <summary>
+        ///  Demands permissions appropriate to the dialog to be shown.
+        /// </summary>
+        protected override void CheckPermissionsToShowDialog()
+        {
+            base.CheckPermissionsToShowDialog();
+        }
+
+        #endregion Protected Methods
+
+        //---------------------------------------------------
+        //
+        // Internal Methods
+        //
+        //---------------------------------------------------
+        #region Internal Methods
+
+        internal override bool RunFileDialog(NativeMethods.OPENFILENAME_I ofn)
+        {
+            // Windows XP is not supported by .NET Core so this path should never execute.
+
+            // An alternative would be to override the RunDialog method and duplicate
+            // System.Windows.Forms.FolderBrowserDialog.RunDialogOld code when UseVista is false.
+
+            throw new PlatformNotSupportedException();
+        }
+
+        internal override void PrepareVistaDialog(IFileDialog dialog)
```

Comment:
Have you tried running this code on a Vista machine?

Reply:
#4029

Reply:
> 
> 
> Have you tried running this code on a Vista machine?

Can we run dotnet core application in Vista?

Reply:
According to the [doc](https://github.com/dotnet/core/blob/master/release-notes/5.0/5.0-supported-os.md#windows), Vista is not supported in .Net 5.

Reply:
It hasn't anything to do with Vista except that since Vista, there is a "modern" file/folder dialog instead of the legacy. So on any Windows >= Vista, including Windows 10, it calls PrepareVistaDialog, just like the existing FileDialog. This is just a wrapper based on FileDialog doing nothing else than setting a flag to make this being a folder dialog instead. **IF** it won't work on Vista machine, then FileDialog won't work either.

https://github.com/dotnet/wpf/blob/d49f8ddb889b5717437d03caa04d7c56819c16aa/src/Microsoft.DotNet.Wpf/src/PresentationFramework/Microsoft/Win32/FileDialog.cs#L817-L829
https://github.com/dotnet/wpf/blob/d49f8ddb889b5717437d03caa04d7c56819c16aa/src/Microsoft.DotNet.Wpf/src/PresentationFramework/Microsoft/Win32/FileDialog.cs#L1835-L1845

Reply:
The reason I haven't submitted a PR myself is that I wasn't able to test the code. Did you manage to run it from an app?

It looks like the only thing you changed is the name - I used `PickFolderFileDialog` because it's a file dialog with a pick folder flag. I think an "open" operation on folder is a bit misleading, what would be the corresponding save folder dialog?

That said, I am not _that_ picky about the name, as long as we get the dialog!

Reply:
@fabiant3 if implementing and testing the legacy dialog would make it eligible for .NET FW I am happy to do that.

Reply:
Yup, pretty much the testing portion is what the concern is. Thank you!